Ginger/GingerCoreNET/RunLib/CLILib/CLIConfigFile.cs

```diff
@@ -37,23 +37,29 @@ public async void Execute(RunsetExecutor runsetExecutor)
         public string CreateContent(Solution solution, RunsetExecutor runsetExecutor, CLIHelper cliHelper)
         {
             string sConfig = null;
-            if (cliHelper.DownloadUpgradeSolutionFromSourceControl == true)
-            {               
-                sConfig = "SourceControlType=" + solution.SourceControl.GetSourceControlType.ToString() + Environment.NewLine;
-                sConfig += "SourceControlUrl=" + solution.SourceControl.SourceControlURL.ToString() + Environment.NewLine;
-                sConfig += "SourceControlUser=" + solution.SourceControl.SourceControlUser.ToString() + Environment.NewLine;
-                sConfig += "SourceControlPassword=" + solution.SourceControl.SourceControlPass.ToString() + Environment.NewLine;
-                if (solution.SourceControl.GetSourceControlType == SourceControlBase.eSourceControlType.GIT && solution.SourceControl.SourceControlProxyAddress.ToLower().ToString() == "true")
+            if (solution != null)
+            {
+                if (cliHelper.DownloadUpgradeSolutionFromSourceControl == true)
                 {
-                    sConfig += "SourceControlProxyServer=" + solution.SourceControl.SourceControlProxyAddress.ToString() + Environment.NewLine;
-                    sConfig += "SourceControlProxyPort=" + solution.SourceControl.SourceControlProxyPort.ToString() + Environment.NewLine;
+                    sConfig = "SourceControlType=" + solution.SourceControl.GetSourceControlType.ToString() + Environment.NewLine;
+                    sConfig += "SourceControlUrl=" + solution.SourceControl.SourceControlURL.ToString() + Environment.NewLine;
+                    sConfig += "SourceControlUser=" + solution.SourceControl.SourceControlUser.ToString() + Environment.NewLine;
+                    sConfig += "SourceControlPassword=" + solution.SourceControl.SourceControlPass.ToString() + Environment.NewLine;
+                    if (solution.SourceControl.SourceControlProxyAddress != null)
```

Comment:
shift If's + add the validation for solution.SourceControl.SourceControlProxyAddress.ToLower().ToString() == "true" after checking it is not null